### PR TITLE
[CHAD-8882] Added holdTime preference to dimming-remote devices

### PR DESCRIPTION
**Pull request by dsobierajsk**

### CODE REVIEW

**Review by KKlimczukS — APPROVED** [review]
lgtm
@greens

File: drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua [hunk 1]
```diff
@@ -37,14 +37,32 @@ local function can_handle_zigbee_dimming_remote(opts, driver, device, ...)
...
+    local comp = device.profile.components[string.format("button%d", button_number)]
+
+    device:emit_event(event)
+    device:emit_component_event(comp, event)
+  end
+end
+
+local function button_hold_handler(button_number)
+  return function(driver, device, value, zb_rx)
+    device:set_field(string.format(BUTTON_X_HOLD, button_number), true, { persist = true })
```
Comment (greens) [inline drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua:51]: this shouldn't need to use persistent storage, considering the timeframe

**Review by greens — COMMENTED** [review]
I'm in favor of keeping the button_utils version because of the timeout threshold. If we miss a released event I don't want us sending a false held event later.

File: drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua [hunk 2]
```diff
@@ -37,14 +38,44 @@ local function can_handle_zigbee_dimming_remote(opts, driver, device, ...)
...
+local function send_held_event_if_applicable(device, button_number)
+  local event = capabilities.button.button.held({ state_change = true })
+  local comp = device.profile.components[string.format("button%d", button_number)]
+  local button_hold_time = get_button_hold_time(device, button_number)
+
+  device:set_field(string.format(BUTTON_X_HOLD, button_number), nil)
+
+  if button_hold_time and button_hold_time < TIMEOUT_THRESHOLD then
+    device:emit_event(event)
+    device:emit_component_event(comp, event)
```
Comment (greens) [inline drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua:72]: Shouldn't we send a `press` event if the hold time is less than some period of time?

Comment (dsobierajsk) [reply drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua:72]: Imo pushed event should be sent only when OnOff message is received. Device recognizes itself if button was hold or pushed only

Comment (greens) [reply drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua:72]: Looks like the DTH sends a pushed if it's not a hold, unless I'm reading wrong?

We want to be sending whatever the DTH would have sent.

Comment (dsobierajsk) [reply drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua:72]: > Looks like the DTH sends a pushed if it's not a hold, unless I'm reading wrong?

Yes, it is

Comment (greens) [reply drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua:72]: so you'll be making a change here to send a pushed if the hold is shorter than the held threshold?

**Comment by dsobierajsk** [comment]
Hi @varzac, could you look at this too? Thanks

File: drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua (hunk 2 above)
Comment (greens) [reply drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua:72]: I think right now we want to copy DTH functionality.

Comment (dsobierajsk) [reply drivers/SmartThings/zigbee-button/src/dimming-remote/init.lua]: Idk why, but button_utils started working properly. I've noticed that device should have also holdTime preference so I added it and rollbacked previous changes.

**Review by KKlimczukS — APPROVED** [review]

File: drivers/SmartThings/zigbee-button/fingerprints.yml [hunk 3]
```diff
@@ -108,12 +108,12 @@ zigbeeManufacturer:
...
     manufacturer: OSRAM
     model: LIGHTIFY Dimming Switch
-    deviceProfileName: two-buttons-battery
+    deviceProfileName: osram-two-buttons-battery
   - id: CentraLite/3130
     deviceLabel: Centralite Button
     manufacturer: CentraLite
     model: 3130
-    deviceProfileName: two-buttons-battery
+    deviceProfileName: osram-two-buttons-battery
```
[the remark this answers is not in the document]
Comment (dsobierajsk) [reply drivers/SmartThings/zigbee-button/fingerprints.yml:116]: It doesn't matter. I have a CentraLite device and it is branded as OSRAM, but you might be right and this name should be better.

**Review by greens — APPROVED** [review]

**Review by varzac — APPROVED** [review]

**Pull request merged**